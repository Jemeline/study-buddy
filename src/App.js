import React, {useState,useEffect} from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import { BrowserRouter, Link} from 'react-router-dom';
import Routes from './routes/routes';
import {getLoginStatus} from './utils/common';
import Header from './components/Navigation/Header.component';
import NavDrawer from './components/Navigation/NavDrawer.component';


function App(){
  document.title = "Study Buddy";
  const [isLoggedIn, setIsLoggedIn] = useState(getLoginStatus());
  const [isOpenDrawer, setIsOpenDrawer] = useState(false); 
  const [isHome, setIsHome] = useState(false);
  
  
  return (
    <div className="App" data-testid="App">
      <BrowserRouter>
        <Header isLoggedIn={isLoggedIn} isHome={isHome} setIsLoggedIn={setIsLoggedIn} setIsOpenDrawer={setIsOpenDrawer}></Header>
        <NavDrawer isOpenDrawer={isOpenDrawer} isHome={isHome} setIsOpenDrawer={setIsOpenDrawer} isLoggedIn={isLoggedIn}/>
        <div> 
          <Routes setIsLoggedIn={setIsLoggedIn} setIsHome={setIsHome}/>
        </div>
      </BrowserRouter>
    </div>
  );
};

export default App;