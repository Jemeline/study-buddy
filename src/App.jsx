import React from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import Routes from './routes/Routes.jsx';
import { Button } from 'react-bootstrap';
import { withFirebase } from './utils/Firebase';

const App = ({ Firebase }) => {
  const logout = async () => await Firebase.logout();

  return (
    <div className="App">
      <Router>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/login">Login</Link></li>
        <li><Link to="/register">Register</Link></li>
        <li><Link to="/recover">Recover password</Link></li>
        <li><Link to="/protectedroute">Protected Component (Protected)</Link></li>
        <li><Link to="/student/dashboard">Student Dashboard (Student Claim)</Link></li>
        <li><Link to="/tutor/dashboard">Tutor Dashboard (Tutor Claim)</Link></li>
        <li><Link to="/admin/dashboard">Admin Dashboard (Admin Claim)</Link></li>
        <li><Link to="/admin/newAdmin">New Admin (Admin Claim)</Link></li>
        <Button as={Link} onClick={logout} to="/login">Logout</Button>
        <Routes />
      </Router>
    </div>
  );
};

export default withFirebase(App);