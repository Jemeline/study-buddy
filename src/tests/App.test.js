import { render, screen, cleanup } from '@testing-library/react';
import App from '../App';

test('renders App component', () => {
  render(<App />);
  const element = screen.getByTestId("App");
  expect(element).toBeInTheDocument();
});

afterEach(cleanup)
 
it('should take a snapshot', () => {
    const { asFragment } = render(<App />)
    
    expect(asFragment(<App />)).toMatchSnapshot()
});