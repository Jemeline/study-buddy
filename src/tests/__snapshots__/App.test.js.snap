// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="App"
  >
    <div
      hidden=""
      style="flex-grow: 1;"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        style="background: rgb(109, 174, 219); height: 65px;"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
            style="background: rgb(109, 174, 219);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="flex-grow: 1; text-align: left;"
          >
            Study Buddy
          </h6>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            hidden=""
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            as="[object Object]"
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            hidden=""
            style="color: rgb(255, 255, 255);"
            tabindex="0"
            to="/auth"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sign Out
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
    <div
      hidden=""
    />
    <div>
      <div>
        <h4>
          Home
        </h4>
        <p>
          This is the Home Page.
        </p>
        <a
          href="/auth"
        >
          Login/Register
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
